src/openvpn/multi.h

```diff
@@ -623,6 +627,35 @@ multi_process_outgoing_tun(struct multi_context *m, const unsigned int mpp_flags
            mi->context.c2.to_tun.len);
 #endif
     set_prefix(mi);
+#ifdef ENABLE_VLAN_TAGGING
+    if (m->top.options.vlan_accept == VAF_ONLY_UNTAGGED_OR_PRIORITY)
+    {
+        /* Packets aren't VLAN-tagged on the tap device.  */
+
+        if (m->top.options.vlan_pvid != mi->context.options.vlan_pvid)
```

Comment:
if in the config we set VAF_ONLY_UNTAGGED_OR_PRIORITY, but then we also set vlan-pvid to some X, what happens? I am wondering if this check could break under some (bogus) configurations.

Reply:
Setting the global vlan-pvid to some X is a perfectly valid use-case when globally setting vlan-accept to VAF_ONLY_UNTAGGED_OR_PRIORITY. You can imagine the tap device representing a port on the "openvpn smart switch" - same as each connected client also represents a port. The PVID is the port's VID. In untagged mode, the port only outputs packets matching the configured PVID. If the VID does not match, the packet is not forwarded to the tap device. If I'm not mistaken, that's the worst that can happen ... Did you have specific bogus configuration scenario in mind?